src/plugins/xsOverlay.desktop/index.ts

```diff
@@ -0,0 +1,287 @@
+                }
+
+                // make emotes and channel mentions readable
+                const emoteMatches = finalMsg.match(new RegExp("(<a?:\\w+:\\d+>)", "g"));
+                const channelMatches = finalMsg.match(new RegExp("<(#\\d+)>", "g"));
+
+                if (emoteMatches) {
+                    for (const eMatch of emoteMatches) {
+                        finalMsg = finalMsg.replace(new RegExp(`${eMatch}`, "g"), `:${eMatch.split(":")[1]}:`);
+                    }
+                }
+
+                if (channelMatches) {
+                    for (const cMatch of channelMatches) {
+                        let channelId = cMatch.split("<#")[1];
+                        channelId = channelId.substring(0, channelId.length - 1);
+                        finalMsg = finalMsg.replace(new RegExp(`${cMatch}`, "g"), `<b><color=#${channelPingColor}>#${ChannelStore.getChannel(channelId).name}</color></b>`);
+                    }
+                }
+
+                sendMsgNotif(titleString, finalMsg, message);
+            } catch (err) {
+                XSLog.error(`Failed to catch MESSAGE_CREATE: ${err}`);
+            }
+        }
+    }
+});
+
+function sendMsgNotif(titleString: string, content: string, message: Message) {
+    fetch(`https://cdn.discordapp.com/avatars/${message.author.id}/${message.author.avatar}.png?size=128`).then(response => response.arrayBuffer()).then(result => {
+        const msgData = {
+            messageType: 1,
+            index: 0,
+            timeout: settings.store.timeout,
+            height: calculateHeight(cleanMessage(content)),
+            opacity: settings.store.opacity,
+            volume: settings.store.volume,
+            audioPath: settings.store.soundPath,
+            title: titleString,
+            content: content,
+            useBase64Icon: true,
+            icon: result,
+            sourceApp: "Vencord"
+        };
+        Native.sendToOverlay(msgData);
+    });
+}
+
+function sendOtherNotif(content: string, titleString: string) {
+    const msgData = {
+        messageType: 1,
+        index: 0,
+        timeout: settings.store.timeout,
+        height: calculateHeight(cleanMessage(content)),
+        opacity: settings.store.opacity,
+        volume: settings.store.volume,
+        audioPath: settings.store.soundPath,
+        title: titleString,
+        content: content,
+        useBase64Icon: false,
+        icon: null,
+        sourceApp: "Vencord"
+    };
+    Native.sendToOverlay(msgData);
+}
+
+function shouldNotify(message: Message, channel: Channel) {
+    const currentUser = UserStore.getCurrentUser();
+    if (message.author.id === currentUser.id) return false;
+    if (message.author.bot && settings.store.ignoreBots) return false;
+    if (MuteStore.allowAllMessages(channel) || message.mention_everyone && !MuteStore.isSuppressEveryoneEnabled(message.guild_id)) return true;
+
+    return message.mentions.some(m => m.id === currentUser.id);
```

Comment:
i think just 
```suggestion
    return message.mentioned;
```
will work?